src/collections/config/types/vectorizer.ts

```diff
@@ -406,6 +407,20 @@ export type Text2VecVoyageAIConfig = {
   vectorizeCollectionName?: boolean;
 };
 
+/**
+ * The configuration for text vectorization using Weaviate's self-hosted text-based embedding models.
+ *
+ * TODO: add documentation reference once available.
+ */
+export interface Text2VecWeaviateConfig {
```

Comment:
Can we change this declaration to be `type` rather than `interface` to maintain the convention of the codebase?